### PR TITLE
fix serial subscribe topic mismatch

### PR DESCRIPTION
**Pull request by kgbu**
The serial device subscribe topic:

USED TO BE: <topicPrefix>/<gatewayName>/<deviceName>/subscribe
SHOULD BE: <topicPrefix>/<gatewayName>/<deviceName>/<type>/subscribe

Because of this mismatch, we could not output message data of subscribed topic to serial port.

Subscribe topic registration and validation codes are fixed.

### CODE REVIEW

File: broker/subscribe.go [hunk 1]
```diff
@@ -64,12 +66,12 @@ func (s Subscribed) Delete(topic string) error {
...
-	t := strings.Join([]string{b.TopicPrefix, b.GatewayName, deviceName, "subscribe"}, "/")
+func (b *Broker) AddSubscribed(deviceTopic message.TopicString, qos byte) error {
+	t := strings.Join([]string{b.TopicPrefix, b.GatewayName, deviceTopic.Str}, "/")
 	log.Infof("subscribe: %#v", t)
 	return b.Subscribed.Add(t, qos)
 }
-func (b *Broker) DeleteSubscribed(deviceName string, qos byte) error {
-	t := strings.Join([]string{b.TopicPrefix, b.GatewayName, deviceName}, "/")
+func (b *Broker) DeleteSubscribed(deviceTopic message.TopicString, qos byte) error {
+	t := strings.Join([]string{b.TopicPrefix, b.GatewayName, deviceTopic.String()}, "/")
```
Comment (shirou) [inline broker/subscribe.go]: Should `String()` be `Str`?

File: device/dummy.go [hunk 2]
```diff
@@ -139,7 +143,7 @@ func (device DummyDevice) MainLoop(channel chan message.Message) error {
 			}
 			channel <- msg
 		case msg, _ := <-device.DeviceChan.Chan:
-			if !strings.HasSuffix(msg.Topic, device.Name) {
+			if !strings.HasSuffix(msg.Topic, device.SubscribeTopic.Str) {
```
Comment (shirou) [inline device/dummy.go]: If `subscribe` is not set, `device.SubscribeTopic` is nil. So, this code will panic.  Yes, I know if not `subscribe`d, none of messages will come from `device.DeviceChan.Chan`, but someone change some codes.
So, my suggestion is,
1. Initialize device.SubscribeTopic to something
2. check device.SubscribeTopic is not nil.
I prefer 2, but you may choose one of or any other.

File: broker/subscribe.go (hunk 1 above)
Comment (kgbu) [inline broker/subscribe.go]: Yes. I forgot to change.
 I'll fix it.

File: device/dummy.go (hunk 2 above)
Comment (kgbu) [inline device/dummy.go]: I also prefer 2 (check nil or not).
I'll change it.
Thanks for your good suggestion.

Comment (shirou) [inline device/dummy.go]: How about to add checking SubscribeTopic.Str like `device.SubscribeTopic.Str != "" && !strings.HasSuffix...` ?

**Comment by shirou** [comment]
LGTM!

**Pull request merged**